### PR TITLE
Fix UT for PR#2215

### PR DESCRIPTION
**Pull request by chenzhitong**
Fix UT for PR https://github.com/neo-project/neo/pull/2215

- [x] IO/Caching/UT_DataCache.cs

- [x] IO/Caching/UT_CloneCache.cs

- [x] IO/UT_SerializableWrapper.cs

- [x] Ledger/UT_TransactionVerificationContext.cs

- [x] Ledger/UT_TrimmedBlock.cs

- [x] Network/P2p/Payloads/UT_Header.cs

- [x] Network/P2p/Payloads/UT_Transaction.cs

- [x] Persistence/UT_MemoryStore.cs

- [x] Persistence/UT_ReadOnlyView.cs

- [x] SmartContract/Native/UT_PolicyContract.cs

- [x] SmartContract/Native/UT_NeoToken.cs

- [x] SmartContract/UT_InteropPrices.cs

- [x] SmartContract/UT_InteropService.cs

- [x] SmartContract/UT_InteropService.NEO.cs

- [x] SmartContract/UT_SmartContractHelper.cs

- [x] SmartContract/UT_Syscalls.cs

- [x] UT_DataCache.cs

- [x] Wallets/UT_Wallet.cs

### CODE REVIEW

**Comment by chenzhitong** [comment]
SnapShotCache removes the dynamic type, so how should it be used?

image

image

**Comment by shargon** [comment]
I will help you

File: src/neo/SmartContract/Native/LedgerContract.cs [hunk 1]
```diff
@@ -181,7 +181,7 @@ private Transaction GetTransactionFromBlock(DataCache snapshot, byte[] blockInde
             return GetTransaction(snapshot, block.Hashes[txIndex + 1]);
         }
 
-        private static TrimmedBlock Trim(Block block)
+        internal static TrimmedBlock Trim(Block block)
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/LedgerContract.cs:184]: Why internal?

Comment (shargon) [reply src/neo/SmartContract/Native/LedgerContract.cs]: Fixed

File: src/neo/SmartContract/StorageItem.cs [hunk 2]
```diff
@@ -3,11 +3,12 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 using System.Numerics;
 
 namespace Neo.SmartContract
 {
-    public class StorageItem : ISerializable
+    public class StorageItem : ISerializable, IEquatable<StorageItem>
```
Comment (erikzhang) [inline src/neo/SmartContract/StorageItem.cs:11]: Why?

Comment (erikzhang) [reply src/neo/SmartContract/StorageItem.cs:11]: @shargon

**Comment by chenzhitong** [comment]
Unit test result:
728 Passed
4 Failed

- [ ] Neo.UnitTests.Ledger/UT_TrimmedBlock/TestGetBlock
- [ ] Neo.UnitTests.SmartContract.Native/UT_NeoToken/Check_GetCommittee
- [ ] Neo.UnitTests.SmartContract.Native/UT_NeoToken/TestCheckCandidate
- [ ] Neo.UnitTests.SmartContract.Native/UT_NeoToken/TestClaimGas

File: src/neo/SmartContract/StorageItem.cs (hunk 2 above)
Comment (shargon) [reply src/neo/SmartContract/StorageItem.cs:11]: I will remove it, it was for detect the problem but the checks was fixed

**Review by erikzhang — APPROVED** [review]

**Comment by shargon** [comment]
Wait please, I need to recover `TestBlockchain_GetTransactionHeight`, it was removed

**Review by shargon — APPROVED** [review]

**Pull request merged**